packages/filesystem/src/electron-browser/file-dialog/electron-file-dialog-service.ts

```diff
@@ -35,10 +36,14 @@ import { OpenDialogOptions, SaveDialogOptions } from '../../electron-common/elec
 export class ElectronFileDialogService extends DefaultFileDialogService {
 
     @inject(MessageService) protected readonly messageService: MessageService;
+    @inject(RemoteService) protected readonly remoteService: RemoteService;
 
     override async showOpenDialog(props: OpenFileDialogProps & { canSelectMany: true }, folder?: FileStat): Promise<MaybeArray<URI> | undefined>;
     override async showOpenDialog(props: OpenFileDialogProps, folder?: FileStat): Promise<URI | undefined>;
     override async showOpenDialog(props: OpenFileDialogProps, folder?: FileStat): Promise<MaybeArray<URI> | undefined> {
+        if (this.remoteService.isConnected()) {
```

Comment:
I've changed the dialog so that it no longer needs to know whether it runs on a remote server. I'm not sure that is the best solution, but yours is imo out of scope.